### PR TITLE
Add feature for viewing, editing and filling User Information

### LINKED ISSUE #21: feat: Edit profile of user
### Description
As a User, I should be able to edit information on my profile that is displayed on my dashboard.

## Definition of Done
- [ ] The user is able to edit the profile
- [ ] Approval by 2 mentors.

## Estimation
3 hours

### PR DESCRIPTION
**Pull request by bismitaguha**
### Description

- Add feature for filling User Information
- Add feature of displaying User Information (filled/not filled based on condition)

Fixes #21 

### Type of Change:

- Code
- User Interface

**Code/Quality Assurance Only**
- New feature (non-breaking change which adds functionality pre-approved by mentors)

#### When User information is filled
Screenshot from 2020-06-24 03-26-03
##### When User information is not filled
Screenshot from 2020-06-24 03-26-48

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [x] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by bismitaguha** [comment]
@sidvenu @abha224 I am not able to update the state on already existent information. It might be because of a minor error. Please look into the recent commit!

**Comment by abha224** [comment]
@bismitaguha Pls check out a little about performing React state updates on an unmounted component

File: package.json [hunk 1]
```diff
@@ -42,5 +42,8 @@
       "last 1 firefox version",
       "last 1 safari version"
     ]
+  },
+  "devDependencies": {
```
Comment (abha224) [inline package.json]: outside the scope of this pr

File: src/AuthRoute.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+export const AuthRoute = ({ component: Component, ...rest }) => (
+    <>
+    <Navbar/>
+    <Route {...rest} render={props => (
+        localStorage.getItem('token') === null
+        ? <Component {...props} />
+        : <Redirect to='/'/>
+    )} />
+    </>
+)
```
Comment (sidvenu) [inline src/AuthRoute.js]: Add new line

File: src/actions/login.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        callback()
+    }
+    catch (err) {
+        dispatch({
+            type: REGISTER_ERRORS,
+            payload: err.response.data
+        });
+        callback()
+    }
+};
```
Comment (sidvenu) [inline src/actions/login.js]: Add new line

File: src/actions/user.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+            payload: res.data
+        });
+    }
+    catch (err) {
+        dispatch({
+            type: USER_ERRORS,
+            payload: err.response.data
+        })
+    }
+}
```
Comment (sidvenu) [inline src/actions/user.js]: Add new line

File: src/components/App.js [hunk 5]
```diff
@@ -1,19 +1,28 @@
-import React, { Component } from 'react';
-import { Route, Switch, BrowserRouter as Router } from 'react-router-dom'
-// import Routes from '../Routes'
+import React, { Component } from 'react'
+// import { connect } from 'react-redux'
+import { Route, Switch, Router } from 'react-router-dom'
+import Routes from '../Routes'
+// import Navbar from './Navbar'
+import { history } from '../store'
```
Comment (sidvenu) [inline src/components/App.js]: Why are some imports commented out?

File: src/components/Form.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+export default class Form extends Component {
+    render() {
+        return (
+            <div>
+                forms
+            </div>
+        )
+    }
+}
```
Comment (sidvenu) [inline src/components/Form.js]: Add new line

File: src/components/Login.js [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+
+const mapStateToProps = state => ({
+    login: state.login.login,
+    loginerror: state.login.loginerror
+})
+
+export default connect(
+    mapStateToProps,
+    { postLogin, }
+)(Login)
```
Comment (sidvenu) [inline src/components/Login.js]: Add new line

File: src/components/Menubar.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+};
+
+const mapStateToProps = state => ({
+    userinfo: state.info.userinfo,
+})
+
+export default connect(
+    mapStateToProps,
+    { getInfo, }
+)(Menubar)
```
Comment (sidvenu) [inline src/components/Menubar.js]: Add new line

**Review by sidvenu — CHANGES_REQUESTED** [review]
Remove prettier for this PR, for now, as @abha224 mentioned. And please add new lines to the files that doesn't have it at the end - I tried to make a review comment on each file that doesn't have, but stopped halfway considering the problem exists for many files.

File: src/actions/login.js (hunk 3 above)
Comment (bismitaguha) [reply src/actions/login.js]: Merging and rebasing will work, actually these changes are coming from previous commits which are related to the previous PRs. So these new line problems are coming from those commits.

**Comment by bismitaguha** [comment]
@sidvenu @abha224  those new line problems are from old commits which are related to previous PRs. I have resolved them in the previous PRs, so merging and rebasing will automatically remove them. This was unnecessary commits are getting cluttered. please review only new commits!

**Comment by bismitaguha** [comment]
@sidvenu @abha224 These commits - [8a6ffc3](https://github.com/anitab-org/open-source-programs-web/pull/35/commits/8a6ffc3fe6343eb3519072fb573ae32ce02f9d55), [d219097](https://github.com/anitab-org/open-source-programs-web/pull/35/commits/d219097981f2dbc311421f3db6bf27956cacc54c) & [
3c19ebb](https://github.com/anitab-org/open-source-programs-web/pull/35/commits/3c19ebb189d7e7d354cd0ccd1452e81d54b19f7a) are the only ones concerned with this PR and they do not have any changes related to dependencies or newline errors!

**Comment by sidvenu** [comment]
@bismitaguha there are conflicts in 2 styles due to the merge of #32. Please resolve and force push.

**Review by sidvenu — APPROVED** [review]

**Review by abha224 — APPROVED** [review]

**Pull request merged**